mail/internal/mail/smtp.go

```diff
@@ -199,7 +199,11 @@ func (d *Dialer) DialAndSend(m ...*Message) error {
 	}
 	defer s.Close()
 
-	return Send(s, m...)
+	if errs := Send(s, m...); errs[0] != nil {
```

Comment:
This is only checking if the first message errored. Please check all message results and send the first error found back.